mlrun/utils/condition_evaluator.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import multiprocessing
+import typing
+
+from mlrun.utils import logger
+
+
+def evaluate_condition_in_separate_process(
+    condition: str, context: typing.Dict[str, typing.Any], timeout: int = 5
+):
+
+    if not condition:
+        return True
+
+    receiver, sender = multiprocessing.Pipe()
+    p = multiprocessing.Process(
+        target=_evaluate_condition_wrapper,
+        args=(sender, condition, context),
+    )
+    p.start()
+    if receiver.poll(timeout):
+        result = receiver.recv()
+        p.join()
+        return result
+    else:
+        p.kill()
```

Comment:
Specific reason you're using `kill()` rather than `terminate()`? Are you being aggressive on purpose?